mitodl/mit-open
# Search page cleanup

### What are the relevant tickets?
For #640 

### Description (What does it do?)
This PR makes several changes to the search page:

1. Removes the "Active Facets" display above facet list. ([These chips](https://github.com/mitodl/mit-open/assets/9010790/d38b271a-20e1-43f7-bf6d-8383cc05648b) have been removed.)
2. Now, if a facet is active, its checkbox should always be shown.
     - Facets options are only included in the API results if their result count is nonzero. So if an active facet corresponds to zero results, it previously was not shown. This situation can occur by toggling a facet on, then entering query text. For example: https://mitopen-rc.odl.mit.edu/search/?q=quaternions&t=Visual+Arts
3. The results title count has been removed. (No more '524 results for "Physics".)
4. "Clear all facets" no longer clears the resource tab selection.
5. Adds "Offered by" facet.
6. Facets now use API query names ("department" not "d")
7. Facets now work even if they aren't explicitly included in UI.
    - e.g., `http://localhost:8063/search?platform=mitx` 

### Screenshots (if appropriate):
<img width="1460" alt="Screenshot 2024-03-25 at 3 01 27 PM" src="https://github.com/mitodl/mit-open/assets/9010790/9eae047b-6e9a-4138-a1df-69eee93d52de">

<img width="1622" alt="Screenshot 2024-03-25 at 2 50 42 PM" src="https://github.com/mitodl/mit-open/assets/9010790/d349bbe9-ee59-4b6b-8b4b-f7456a27943b">


### How can this be tested?
1. Visit http://localhost:8063/search
2. Try:
    - click resource tabs
    - entering query text
    - toggling left sidebar filters on and off
    - using the "Clear all" filters button

    Things to look out for
    - URL bar should use "real" API parameter names ("department" not "d")
3. Append a filter not included in the UI to the URL; for example `?platform=mitxonline` or (`&platform=mitxonline`). The results should update.


### Additional Context
